cpp/src/arrow/compute/row/grouper_test.cc

```diff
@@ -898,42 +1064,88 @@ TEST(Grouper, DoubleStringInt64Key) {
   g.ExpectConsume(R"([[-0.0, "be", 7], [0.0, "be", 7]])", "[3, 4]");
 }
 
-TEST(Grouper, RandomInt64Keys) {
-  TestGrouper g({int64()});
+FieldVector AnnotateForRandomGeneration(FieldVector fields) {
+  for (auto& field : fields) {
+    // For each field, constrain random generation to ensure that group ids
+    // can appear more than once.
+    if (is_integer(*field->type())) {
+      field =
+          field->WithMergedMetadata(key_value_metadata({"min", "max"}, {"100", "10000"}));
+    } else if (is_binary_like(*field->type())) {
+      // (note this is unsupported for large binary types)
+      field = field->WithMergedMetadata(key_value_metadata({"unique"}, {"100"}));
+    }
+    field = field->WithMergedMetadata(key_value_metadata({"null_probability"}, {"0.1"}));
+  }
+  return fields;
+}
+
+void TestRandomConsume(TestGrouper g) {
```

Comment:
Ditto, why no cases for `Populate`?

Reply:
`Populate` is exercised below in `TestRandomLookup`, or are suggesting something else?

Reply:
Oh I see. Thank you.